Add features and prepare for the addition of Barrage, Chapter, and Equipment data functions

0t4u opened the pull request:
This PR adds filter and search functions for the ship and equipments classes and prepares for the addition of Barrage, Chapter, and Equipment data functions. These changes are tested. Actually adding Barrage, Chapter, and Equipment functions are in the works. Jest tests are also added for automating tests. Please look into commit messages for more details.

0t4u committed:
feat: get ready for implementing chapter, barrage and voiceline functions

Changed:
- Added barrage, voiceline, chapters to the updater system
- Added barrage. voiceline, chapters sources to SaratogaUtil
- Added barrage, voiceline, chapters to the main Saratoga class
- Changed: how update checking works

0t4u committed:
minor: rebuild docs

Changed:
- Regenerated docma docs
- Minor info addition

0t4u committed:
feat: add filter and search for ships and equipments

Changes:
- Added .filter() and .search() functions for ship data and equipment data
- Regenerate docma docs

0t4u committed:
feat: change version number

Changed:
- version number (i forgot lmao)

0t4u committed:
feat: fix checking file exists

Changed:
- add checks for file exists before accessing file

0t4u committed:
feat: fix updater on first start

Changed:
- how checking for updates works

0t4u committed:
feat: add jest tests

changed:
- add jest test
- add github actions file

0t4u committed:
Update SaratogaValidator.js

Deivu commented:
On the side note, I will merge this once the barrage, chapter and equip data functions are done

0t4u committed:
feat: comply v2 specs

Changes:
 - Comply to: https://docs.google.com/document/d/16rmt309dP2iOlG0tJL2XHgdMGrieUbgmWv-aFVJKmhU/edit
 - Fix jest causing errors in eslint
 - Restructure jest tests

0t4u commented:
Barrage, chapter and equip data functions are soon TM
Idea: Legacy mode. Basically old Saratoga API type of function naming conventions. If you would like this please write a comment here thanks.

0t4u commented:
Ill also fix the non-jest vanilla JS tests soon TM

0t4u commented:
On the side note, there is no eta for barrage, chapter and equip data functions because there are no specs for it yet. If you would like to suggest specs please make changes [this doc](https://docs.google.com/document/d/16rmt309dP2iOlG0tJL2XHgdMGrieUbgmWv-aFVJKmhU/edit?usp=sharing). Thanks!

0t4u committed:
minor: regenerate docma docs

Changes:
 - Regenerated docma docs
 - Change version number

0t4u committed:
legacy api support

Changed:
 - Legacy api support
 - Version number
 - Regenerated docs

0t4u committed:
Add barrage functions

This version adds:
 - Barrage functions
 - Typings for voicelines, barrages, and ships
 - Regenerated docs
 - Partially added JP info

0t4u committed:
Update README.md

0t4u committed:
Fix naming mistake

0t4u committed:
Merge branch 'stopgap' of https://github.com/AzurAPI/Saratoga into stopgap

Deivu commented:
Legacy class can be removed later
